### PR TITLE
22. Subscription- Cleanup and minor updates

### PR DESCRIPTION
**Pull request by afterxleep**
Task/Issue URL: https://app.asana.com/0/72649045549333/1206350282230222/f

**Description**:
Streamlines and cleans up implementation pre-release, incluiding:

**New functionalities**
- Adds link to FAQ
- Added a debouncer for Pixels (to avoid duplicates)
- Improved presentation flow and navigation
- Back button is now properly hidden when user should not navigate back
- User can now navigate back in the webview when adding an email to an account
- Fixes ghost view in Settings
- Few copy changes

**Other improvements**
- Re-organize UI Views code for clarity 
- Improves Deeplinking mechanism in Settings
- Encapsulates State into a single ViewModel property everywhere, and make it read only
- Migrates some subFeature combine observers to callbacks
- Ensures proper Main thread use everywhere
 
**Steps to test this PR**:
1. Subscribe via Apple
2. Remove from device
3. Restore via apple
4. Remove from device
5. Restore a Stripe Subscription

### CODE REVIEW

**Comment by afterxleep** [comment]
@federicocappelli This is the last PR for iOS, with clean-ups and optimizations.

File: DuckDuckGo/Subscription/Subscription.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
+//
+//  SubscriptionModel.swift
```
Comment (miasma13) [inline DuckDuckGo/Subscription/Subscription.swift]: Please update file name to `SubscriptionModel`

File: DuckDuckGo/SettingsView.swift [hunk 2]
```diff
@@ -50,9 +68,76 @@ struct SettingsView: View {
...
+               }
+           }
+       }
+       
+    }
+    
+    // MARK: DeepLink Views
+    @available(iOS 15.0, *)
+    @ViewBuilder
+     func deepLinkdestinationView(for target: SettingsViewModel.SettingsDeepLinkSection) -> some View {
```
Comment (miasma13) [inline DuckDuckGo/SettingsView.swift]: ` deepLinkdestinationView` -> ` deepLink**D**estinationView

File: DuckDuckGo/Subscription/Views/SubscriptionFlowView.swift [hunk 3]
```diff
@@ -209,8 +188,8 @@ struct SubscriptionFlowView: View {
                 title: Text(UserText.subscriptionFoundTitle),
                 message: Text(UserText.subscriptionFoundText),
                 primaryButton: .cancel(Text(UserText.subscriptionFoundCancel)) {
-                    viewModel.transactionError = nil
-                    viewModel.finalizeSubscriptionFlow()
+                    viewModel.clearTransactionError()
+                     viewModel.finalizeSubscriptionFlow()
```
Comment (miasma13) [inline DuckDuckGo/Subscription/Views/SubscriptionFlowView.swift:192]: Space

**Review by miasma13 — APPROVED** [review]

**Pull request merged**